OMCompiler/Compiler/BackEnd/Tearing.mo

```diff
@@ -1462,6 +1463,19 @@ algorithm
   end match;
 end solvable;
 
+protected function isEntrySolved
+  input BackendDAE.AdjacencyMatrixElementEnhancedEntry ent;
+  output Boolean b;
+algorithm
+  b := match ent
+    case (_, BackendDAE.SOLVABILITY_SOLVED(), _) then true;
+    case (_, BackendDAE.SOLVABILITY_PARAMETER(b=b), _) algorithm
```

Comment:
Here I think we are only interested in SOLVED variables. Not SOLVABLE ones. That is we are looking for: the algorithm outputs (assigned vars). I have put an error for the SOLVABILITY_PARAMETER because I did not understand what it was and wanted to see something fail for it to investigate further.

The concept of non-assigned variables of an algorithm **being marked as solvable** is what makes the non-enhanced adjacency matrix problematic. 

Non assigned variables in an algorithm section should not be considered solvable if we are not actually inverting the whole algorithm and solving it as a non-linear system.  We can't really rearrange one statement in a an algorithm section to solve for a specific variable.

Reply:
Yes right now it is used for algorithms only. Because we mark non-output (non-assigned) vars in algorithm statements as SOLVABLE (but not SOLVED). The idea of such a var being considered solvable (individually without treating the whole algorithm as its own NLS) is open for discussion :)

But in any case the function is used to filter out SOLVED variables. It is named `isEntrySolved()`. There is another one called `isEntrySolvable()` that I used in other places. It depends which one you use based on what you need I think.